internal/scheduler/scheduler.go

```diff
@@ -286,35 +286,21 @@ func (s *Scheduler) cycle(ctx *armadacontext.Context, updateAll bool, leaderToke
 }
 
 func (s *Scheduler) updateMetricsFromSchedulerResult(ctx *armadacontext.Context, overallSchedulerResult SchedulerResult) error {
-	for _, job := range overallSchedulerResult.ScheduledJobs {
-		if err := s.schedulerMetrics.Update(
-			ctx,
-			jobdb.JobStateTransitions{
-				Job:       job.(*jobdb.Job),
-				Scheduled: true,
-			},
-			nil,
-		); err != nil {
+	for _, jctx := range overallSchedulerResult.ScheduledJobs {
+		if err := s.schedulerMetrics.UpdateScheduled(jctx); err != nil {
 			return err
 		}
 	}
-	for _, job := range overallSchedulerResult.PreemptedJobs {
-		if err := s.schedulerMetrics.Update(
-			ctx,
-			jobdb.JobStateTransitions{
-				Job:       job.(*jobdb.Job),
-				Preempted: true,
-			},
-			nil,
-		); err != nil {
+	for _, jctx := range overallSchedulerResult.PreemptedJobs {
+		if err := s.schedulerMetrics.UpdatePreempted(jctx); err != nil {
 			return err
 		}
 	}
-	for _, job := range overallSchedulerResult.FailedJobs {
+	for _, jctx := range overallSchedulerResult.FailedJobs {
```

Comment:
The "failed" case is also needed by `UpdateMany` (where we only have jobs, not job scheduling contexts), so it still goes through `Update` (unlike "scheduled" and "preempted").